# HG changeset patch
# User Rik <rik@octave.org>
# Date 1498088666 25200
#      Wed Jun 21 16:44:26 2017 -0700
# Node ID 94a4ba7b0d94020f83f03a91ddf2c4a0050f4c45
# Parent  a6eef0626b2b55807161e1a08b4200414651d921
Remove legacy Fortran code for special functions now handled by std library.

* liboctave/external/slatec-fn/module.mk: Remove acosh, asinh, atanh, erf, erfc
related files from build system.  Remove special rules that generated derfc.f
and erfc.f and that used subst-f77-isnan-macro.sh

* configure.ac: Remove build-aux/subst-f77-isnan-macro.sh from list of
OCTAVE_CONFIG_MOVE_IF_CHANGE_FILES.

* build-aux/module.mk: Remove rules for subst-f77-isnan-macro.in.sh.

* lo-slatec-proto.h (xacosh, xdacosh, xasinh, xdasinh, xatanh, xdatanh, xerf,
xderf, xderfc, xerfc, xgamma, xdgamma, algams, dlgams):
Remove function prototypes.

* subst-f77-isnan-macro.in.sh, acosh.f, asinh.f, atanh.f, dacosh.f, dasinh.f,
datanh.f, derf.f, derfc.in.f, erf.f, erfc.in.f, xacosh.f, xasinh.f, xatanh.f,
xdacosh.f, xdasinh.f, xdatanh.f, xderf.f, xderfc.f, xerf.f, xerfc.f:
Remove unused files.

diff --git a/build-aux/module.mk b/build-aux/module.mk
--- a/build-aux/module.mk
+++ b/build-aux/module.mk
@@ -13,29 +13,27 @@ EXTRA_DIST += \
   %reldir%/mk-octave-config-h.sh \
   %reldir%/mk-opts.pl \
   %reldir%/mk-version-h.in.sh \
   %reldir%/move-if-change \
   %reldir%/stl_algo.h-fixed \
   %reldir%/subst-config-vals.in.sh \
   %reldir%/subst-cross-config-vals.in.sh \
   %reldir%/subst-default-vals.in.sh \
-  %reldir%/subst-f77-isnan-macro.in.sh \
   %reldir%/subst-script-vals.in.sh \
   %reldir%/update-bug-status.sh
 
 GEN_CONFIG_SHELL = \
   %reldir%/mk-default-qt-settings.sh \
   %reldir%/mk-f77-def.sh \
   %reldir%/mk-mxarray-h.sh \
   %reldir%/mk-version-h.sh \
   %reldir%/subst-config-vals.sh \
   %reldir%/subst-cross-config-vals.sh \
   %reldir%/subst-default-vals.sh \
-  %reldir%/subst-f77-isnan-macro.sh \
   %reldir%/subst-script-vals.sh
 
 $(GEN_CONFIG_SHELL) : %.sh : %.in.sh config.status
 	$(AM_V_GEN)$(SHELL) config.status $@-tmp $@
 
 GEN_CONFIG_INC = \
   oct-conf-post.h
 
diff --git a/build-aux/subst-f77-isnan-macro.in.sh b/build-aux/subst-f77-isnan-macro.in.sh
deleted file mode 100644
--- a/build-aux/subst-f77-isnan-macro.in.sh
+++ /dev/null
@@ -1,25 +0,0 @@
-#! /bin/sh
-#
-# Copyright (C) 2016-2017 John W. Eaton
-#
-# This file is part of Octave.
-#
-# Octave is free software; you can redistribute it and/or modify it
-# under the terms of the GNU General Public License as published by
-# the Free Software Foundation; either version 3 of the License, or
-# (at your option) any later version.
-#
-# Octave is distributed in the hope that it will be useful, but
-# WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with Octave; see the file COPYING.  If not, see
-# <http://www.gnu.org/licenses/>.
-
-: ${SED=@SED@}
-
-F77_ISNAN_MACRO="@F77_ISNAN_MACRO@"
-
-$SED -e "${F77_ISNAN_MACRO}"
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -3158,17 +3158,16 @@ AC_CONFIG_COMMANDS([oct-conf-post.h],
 OCTAVE_CONFIG_MOVE_IF_CHANGE_FILES([
   build-aux/mk-default-qt-settings.sh
   build-aux/mk-f77-def.sh
   build-aux/mk-mxarray-h.sh
   build-aux/mk-version-h.sh
   build-aux/subst-config-vals.sh
   build-aux/subst-cross-config-vals.sh
   build-aux/subst-default-vals.sh
-  build-aux/subst-f77-isnan-macro.sh
   build-aux/subst-script-vals.sh])
 
 AC_OUTPUT
 
 ### Print a summary so that important information isn't missed.
 
 AC_MSG_NOTICE([
 
diff --git a/liboctave/external/slatec-fn/acosh.f b/liboctave/external/slatec-fn/acosh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/acosh.f
+++ /dev/null
@@ -1,39 +0,0 @@
-*DECK ACOSH
-      FUNCTION ACOSH (X)
-C***BEGIN PROLOGUE  ACOSH
-C***PURPOSE  Compute the arc hyperbolic cosine.
-C***LIBRARY   SLATEC (FNLIB)
-C***CATEGORY  C4C
-C***TYPE      SINGLE PRECISION (ACOSH-S, DACOSH-D, CACOSH-C)
-C***KEYWORDS  ACOSH, ARC HYPERBOLIC COSINE, ELEMENTARY FUNCTIONS, FNLIB,
-C             INVERSE HYPERBOLIC COSINE
-C***AUTHOR  Fullerton, W., (LANL)
-C***DESCRIPTION
-C
-C ACOSH(X) computes the arc hyperbolic cosine of X.
-C
-C***REFERENCES  (NONE)
-C***ROUTINES CALLED  R1MACH, XERMSG
-C***REVISION HISTORY  (YYMMDD)
-C   770401  DATE WRITTEN
-C   890531  Changed all specific intrinsics to generic.  (WRB)
-C   890531  REVISION DATE from Version 3.2
-C   891214  Prologue converted to Version 4.0 format.  (BAB)
-C   900315  CALLs to XERROR changed to CALLs to XERMSG.  (THJ)
-C   900326  Removed duplicate information from DESCRIPTION section.
-C           (WRB)
-C***END PROLOGUE  ACOSH
-      SAVE ALN2,XMAX
-      DATA ALN2 / 0.6931471805 5994530942E0/
-      DATA XMAX /0./
-C***FIRST EXECUTABLE STATEMENT  ACOSH
-      IF (XMAX.EQ.0.) XMAX = 1.0/SQRT(R1MACH(3))
-C
-      IF (X .LT. 1.0) CALL XERMSG ('SLATEC', 'ACOSH', 'X LESS THAN 1',
-     +   1, 2)
-C
-      IF (X.LT.XMAX) ACOSH = LOG (X + SQRT(X*X-1.0))
-      IF (X.GE.XMAX) ACOSH = ALN2 + LOG(X)
-C
-      RETURN
-      END
diff --git a/liboctave/external/slatec-fn/asinh.f b/liboctave/external/slatec-fn/asinh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/asinh.f
+++ /dev/null
@@ -1,74 +0,0 @@
-*DECK ASINH
-      FUNCTION ASINH (X)
-C***BEGIN PROLOGUE  ASINH
-C***PURPOSE  Compute the arc hyperbolic sine.
-C***LIBRARY   SLATEC (FNLIB)
-C***CATEGORY  C4C
-C***TYPE      SINGLE PRECISION (ASINH-S, DASINH-D, CASINH-C)
-C***KEYWORDS  ARC HYPERBOLIC SINE, ASINH, ELEMENTARY FUNCTIONS, FNLIB,
-C             INVERSE HYPERBOLIC SINE
-C***AUTHOR  Fullerton, W., (LANL)
-C***DESCRIPTION
-C
-C ASINH(X) computes the arc hyperbolic sine of X.
-C
-C Series for ASNH       on the interval  0.          to  1.00000D+00
-C                                        with weighted error   2.19E-17
-C                                         log weighted error  16.66
-C                               significant figures required  15.60
-C                                    decimal places required  17.31
-C
-C***REFERENCES  (NONE)
-C***ROUTINES CALLED  CSEVL, INITS, R1MACH
-C***REVISION HISTORY  (YYMMDD)
-C   770401  DATE WRITTEN
-C   890531  Changed all specific intrinsics to generic.  (WRB)
-C   890531  REVISION DATE from Version 3.2
-C   891214  Prologue converted to Version 4.0 format.  (BAB)
-C***END PROLOGUE  ASINH
-      DIMENSION ASNHCS(20)
-      LOGICAL FIRST
-      SAVE ALN2, ASNHCS, NTERMS, XMAX, SQEPS, FIRST
-      DATA ALN2 /0.6931471805 5994530942E0/
-      DATA ASNHCS( 1) /   -.1282003991 1738186E0 /
-      DATA ASNHCS( 2) /   -.0588117611 89951768E0 /
-      DATA ASNHCS( 3) /    .0047274654 32212481E0 /
-      DATA ASNHCS( 4) /   -.0004938363 16265361E0 /
-      DATA ASNHCS( 5) /    .0000585062 07058557E0 /
-      DATA ASNHCS( 6) /   -.0000074669 98328931E0 /
-      DATA ASNHCS( 7) /    .0000010011 69358355E0 /
-      DATA ASNHCS( 8) /   -.0000001390 35438587E0 /
-      DATA ASNHCS( 9) /    .0000000198 23169483E0 /
-      DATA ASNHCS(10) /   -.0000000028 84746841E0 /
-      DATA ASNHCS(11) /    .0000000004 26729654E0 /
-      DATA ASNHCS(12) /   -.0000000000 63976084E0 /
-      DATA ASNHCS(13) /    .0000000000 09699168E0 /
-      DATA ASNHCS(14) /   -.0000000000 01484427E0 /
-      DATA ASNHCS(15) /    .0000000000 00229037E0 /
-      DATA ASNHCS(16) /   -.0000000000 00035588E0 /
-      DATA ASNHCS(17) /    .0000000000 00005563E0 /
-      DATA ASNHCS(18) /   -.0000000000 00000874E0 /
-      DATA ASNHCS(19) /    .0000000000 00000138E0 /
-      DATA ASNHCS(20) /   -.0000000000 00000021E0 /
-      DATA FIRST /.TRUE./
-C***FIRST EXECUTABLE STATEMENT  ASINH
-      IF (FIRST) THEN
-         NTERMS = INITS (ASNHCS, 20, 0.1*R1MACH(3))
-         SQEPS = SQRT (R1MACH(3))
-         XMAX = 1.0/SQEPS
-      ENDIF
-      FIRST = .FALSE.
-C
-      Y = ABS(X)
-      IF (Y.GT.1.0) GO TO 20
-C
-      ASINH = X
-      IF (Y.GT.SQEPS) ASINH = X*(1.0 + CSEVL (2.*X*X-1., ASNHCS,NTERMS))
-      RETURN
-C
- 20   IF (Y.LT.XMAX) ASINH = LOG (Y + SQRT(Y**2+1.))
-      IF (Y.GE.XMAX) ASINH = ALN2 + LOG(Y)
-      ASINH = SIGN (ASINH, X)
-C
-      RETURN
-      END
diff --git a/liboctave/external/slatec-fn/atanh.f b/liboctave/external/slatec-fn/atanh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/atanh.f
+++ /dev/null
@@ -1,78 +0,0 @@
-*DECK ATANH
-      FUNCTION ATANH (X)
-C***BEGIN PROLOGUE  ATANH
-C***PURPOSE  Compute the arc hyperbolic tangent.
-C***LIBRARY   SLATEC (FNLIB)
-C***CATEGORY  C4C
-C***TYPE      SINGLE PRECISION (ATANH-S, DATANH-D, CATANH-C)
-C***KEYWORDS  ARC HYPERBOLIC TANGENT, ATANH, ELEMENTARY FUNCTIONS,
-C             FNLIB, INVERSE HYPERBOLIC TANGENT
-C***AUTHOR  Fullerton, W., (LANL)
-C***DESCRIPTION
-C
-C ATANH(X) computes the arc hyperbolic tangent of X.
-C
-C Series for ATNH       on the interval  0.          to  2.50000D-01
-C                                        with weighted error   6.70E-18
-C                                         log weighted error  17.17
-C                               significant figures required  16.01
-C                                    decimal places required  17.76
-C
-C***REFERENCES  (NONE)
-C***ROUTINES CALLED  CSEVL, INITS, R1MACH, XERMSG
-C***REVISION HISTORY  (YYMMDD)
-C   770401  DATE WRITTEN
-C   890531  Changed all specific intrinsics to generic.  (WRB)
-C   890531  REVISION DATE from Version 3.2
-C   891214  Prologue converted to Version 4.0 format.  (BAB)
-C   900315  CALLs to XERROR changed to CALLs to XERMSG.  (THJ)
-C   900326  Removed duplicate information from DESCRIPTION section.
-C           (WRB)
-C***END PROLOGUE  ATANH
-      DIMENSION ATNHCS(15)
-      LOGICAL FIRST
-      SAVE ATNHCS, NTERMS, DXREL, SQEPS, FIRST
-      DATA ATNHCS( 1) /    .0943951023 93195492E0 /
-      DATA ATNHCS( 2) /    .0491984370 55786159E0 /
-      DATA ATNHCS( 3) /    .0021025935 22455432E0 /
-      DATA ATNHCS( 4) /    .0001073554 44977611E0 /
-      DATA ATNHCS( 5) /    .0000059782 67249293E0 /
-      DATA ATNHCS( 6) /    .0000003505 06203088E0 /
-      DATA ATNHCS( 7) /    .0000000212 63743437E0 /
-      DATA ATNHCS( 8) /    .0000000013 21694535E0 /
-      DATA ATNHCS( 9) /    .0000000000 83658755E0 /
-      DATA ATNHCS(10) /    .0000000000 05370503E0 /
-      DATA ATNHCS(11) /    .0000000000 00348665E0 /
-      DATA ATNHCS(12) /    .0000000000 00022845E0 /
-      DATA ATNHCS(13) /    .0000000000 00001508E0 /
-      DATA ATNHCS(14) /    .0000000000 00000100E0 /
-      DATA ATNHCS(15) /    .0000000000 00000006E0 /
-      DATA FIRST /.TRUE./
-C***FIRST EXECUTABLE STATEMENT  ATANH
-      IF (FIRST) THEN
-         NTERMS = INITS (ATNHCS, 15, 0.1*R1MACH(3))
-         DXREL = SQRT (R1MACH(4))
-         SQEPS = SQRT (3.0*R1MACH(3))
-      ENDIF
-      FIRST = .FALSE.
-C
-      Y = ABS(X)
-      IF (Y .GE. 1.0) THEN
-         IF (Y .GT. 1.0) THEN
-            ATANH = (X - X) / (X - X)
-         ELSE
-            ATANH = X / 0.0
-         ENDIF
-         RETURN
-      ENDIF
-C
-      IF (1.0-Y .LT. DXREL) CALL XERMSG ('SLATEC', 'ATANH',
-     +   'ANSWER LT HALF PRECISION BECAUSE ABS(X) TOO NEAR 1', 1, 1)
-C
-      ATANH = X
-      IF (Y.GT.SQEPS .AND. Y.LE.0.5) ATANH = X*(1.0 + CSEVL (8.*X*X-1.,
-     1  ATNHCS, NTERMS))
-      IF (Y.GT.0.5) ATANH = 0.5*LOG((1.0+X)/(1.0-X))
-C
-      RETURN
-      END
diff --git a/liboctave/external/slatec-fn/dacosh.f b/liboctave/external/slatec-fn/dacosh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/dacosh.f
+++ /dev/null
@@ -1,40 +0,0 @@
-*DECK DACOSH
-      DOUBLE PRECISION FUNCTION DACOSH (X)
-C***BEGIN PROLOGUE  DACOSH
-C***PURPOSE  Compute the arc hyperbolic cosine.
-C***LIBRARY   SLATEC (FNLIB)
-C***CATEGORY  C4C
-C***TYPE      DOUBLE PRECISION (ACOSH-S, DACOSH-D, CACOSH-C)
-C***KEYWORDS  ACOSH, ARC HYPERBOLIC COSINE, ELEMENTARY FUNCTIONS, FNLIB,
-C             INVERSE HYPERBOLIC COSINE
-C***AUTHOR  Fullerton, W., (LANL)
-C***DESCRIPTION
-C
-C DACOSH(X) calculates the double precision arc hyperbolic cosine for
-C double precision argument X.  The result is returned on the
-C positive branch.
-C
-C***REFERENCES  (NONE)
-C***ROUTINES CALLED  D1MACH, XERMSG
-C***REVISION HISTORY  (YYMMDD)
-C   770601  DATE WRITTEN
-C   890531  Changed all specific intrinsics to generic.  (WRB)
-C   890531  REVISION DATE from Version 3.2
-C   891214  Prologue converted to Version 4.0 format.  (BAB)
-C   900315  CALLs to XERROR changed to CALLs to XERMSG.  (THJ)
-C***END PROLOGUE  DACOSH
-      DOUBLE PRECISION X, DLN2, XMAX,  D1MACH
-      SAVE DLN2, XMAX
-      DATA DLN2 / 0.6931471805 5994530941 7232121458 18 D0 /
-      DATA XMAX / 0.D0 /
-C***FIRST EXECUTABLE STATEMENT  DACOSH
-      IF (XMAX.EQ.0.D0) XMAX = 1.0D0/SQRT(D1MACH(3))
-C
-      IF (X .LT. 1.D0) CALL XERMSG ('SLATEC', 'DACOSH',
-     +   'X LESS THAN 1', 1, 2)
-C
-      IF (X.LT.XMAX) DACOSH = LOG (X+SQRT(X*X-1.0D0))
-      IF (X.GE.XMAX) DACOSH = DLN2 + LOG(X)
-C
-      RETURN
-      END
diff --git a/liboctave/external/slatec-fn/dasinh.f b/liboctave/external/slatec-fn/dasinh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/dasinh.f
+++ /dev/null
@@ -1,89 +0,0 @@
-*DECK DASINH
-      DOUBLE PRECISION FUNCTION DASINH (X)
-C***BEGIN PROLOGUE  DASINH
-C***PURPOSE  Compute the arc hyperbolic sine.
-C***LIBRARY   SLATEC (FNLIB)
-C***CATEGORY  C4C
-C***TYPE      DOUBLE PRECISION (ASINH-S, DASINH-D, CASINH-C)
-C***KEYWORDS  ARC HYPERBOLIC SINE, ASINH, ELEMENTARY FUNCTIONS, FNLIB,
-C             INVERSE HYPERBOLIC SINE
-C***AUTHOR  Fullerton, W., (LANL)
-C***DESCRIPTION
-C
-C DASINH(X) calculates the double precision arc hyperbolic
-C sine for double precision argument X.
-C
-C***REFERENCES  (NONE)
-C***ROUTINES CALLED  D1MACH, DCSEVL, INITDS
-C***REVISION HISTORY  (YYMMDD)
-C   770601  DATE WRITTEN
-C   890531  Changed all specific intrinsics to generic.  (WRB)
-C   890531  REVISION DATE from Version 3.2
-C   891214  Prologue converted to Version 4.0 format.  (BAB)
-C***END PROLOGUE  DASINH
-      DOUBLE PRECISION X, ASNHCS(39), ALN2, SQEPS, XMAX, Y,
-     1  DCSEVL, D1MACH
-      LOGICAL FIRST
-      SAVE ASNHCS, ALN2, NTERMS, XMAX, SQEPS, FIRST
-      DATA ASNHCS(  1) / -.1282003991 1738186343 3721273592 68 D+0     /
-      DATA ASNHCS(  2) / -.5881176118 9951767565 2117571383 62 D-1     /
-      DATA ASNHCS(  3) / +.4727465432 2124815640 7252497560 29 D-2     /
-      DATA ASNHCS(  4) / -.4938363162 6536172101 3601747902 73 D-3     /
-      DATA ASNHCS(  5) / +.5850620705 8557412287 4948352593 21 D-4     /
-      DATA ASNHCS(  6) / -.7466998328 9313681354 7550692171 88 D-5     /
-      DATA ASNHCS(  7) / +.1001169358 3558199265 9661920158 12 D-5     /
-      DATA ASNHCS(  8) / -.1390354385 8708333608 6164722588 86 D-6     /
-      DATA ASNHCS(  9) / +.1982316948 3172793547 3173602371 48 D-7     /
-      DATA ASNHCS( 10) / -.2884746841 7848843612 7472728003 17 D-8     /
-      DATA ASNHCS( 11) / +.4267296546 7159937953 4575149959 07 D-9     /
-      DATA ASNHCS( 12) / -.6397608465 4366357868 7526323096 81 D-10    /
-      DATA ASNHCS( 13) / +.9699168608 9064704147 8782931311 79 D-11    /
-      DATA ASNHCS( 14) / -.1484427697 2043770830 2466583656 96 D-11    /
-      DATA ASNHCS( 15) / +.2290373793 9027447988 0401843789 83 D-12    /
-      DATA ASNHCS( 16) / -.3558839513 2732645159 9789426513 10 D-13    /
-      DATA ASNHCS( 17) / +.5563969408 0056789953 3745390885 54 D-14    /
-      DATA ASNHCS( 18) / -.8746250959 9624678045 6665935201 62 D-15    /
-      DATA ASNHCS( 19) / +.1381524884 4526692155 8688022981 29 D-15    /
-      DATA ASNHCS( 20) / -.2191668828 2900363984 9551422641 49 D-16    /
-      DATA ASNHCS( 21) / +.3490465852 4827565638 3139237068 80 D-17    /
-      DATA ASNHCS( 22) / -.5578578840 0895742439 6301570321 06 D-18    /
-      DATA ASNHCS( 23) / +.8944514661 7134012551 0508827989 33 D-19    /
-      DATA ASNHCS( 24) / -.1438342634 6571317305 5518452394 66 D-19    /
-      DATA ASNHCS( 25) / +.2319181187 2169963036 3261446826 66 D-20    /
-      DATA ASNHCS( 26) / -.3748700795 3314343674 5706045439 99 D-21    /
-      DATA ASNHCS( 27) / +.6073210982 2064279404 5492428800 00 D-22    /
-      DATA ASNHCS( 28) / -.9859940276 4633583177 3701734400 00 D-23    /
-      DATA ASNHCS( 29) / +.1603921745 2788496315 2326382933 33 D-23    /
-      DATA ASNHCS( 30) / -.2613884735 0287686596 7161343999 99 D-24    /
-      DATA ASNHCS( 31) / +.4267084960 6857390833 3581653333 33 D-25    /
-      DATA ASNHCS( 32) / -.6977021703 9185243299 7307733333 33 D-26    /
-      DATA ASNHCS( 33) / +.1142508833 6806858659 8126933333 33 D-26    /
-      DATA ASNHCS( 34) / -.1873529207 8860968933 0210133333 33 D-27    /
-      DATA ASNHCS( 35) / +.3076358441 4464922794 0659200000 00 D-28    /
-      DATA ASNHCS( 36) / -.5057736403 1639824787 0463999999 99 D-29    /
-      DATA ASNHCS( 37) / +.8325075471 2689142224 2133333333 33 D-30    /
-      DATA ASNHCS( 38) / -.1371845728 2501044163 9253333333 33 D-30    /
-      DATA ASNHCS( 39) / +.2262986842 6552784104 1066666666 66 D-31    /
-      DATA ALN2 / 0.6931471805 5994530941 7232121458 18D0 /
-      DATA FIRST /.TRUE./
-C***FIRST EXECUTABLE STATEMENT  DASINH
-      IF (FIRST) THEN
-         NTERMS = INITDS (ASNHCS, 39, 0.1*REAL(D1MACH(3)) )
-         SQEPS = SQRT(D1MACH(3))
-         XMAX = 1.0D0/SQEPS
-      ENDIF
-      FIRST = .FALSE.
-C
-      Y = ABS(X)
-      IF (Y.GT.1.0D0) GO TO 20
-C
-      DASINH = X
-      IF (Y.GT.SQEPS) DASINH = X*(1.0D0 + DCSEVL (2.D0*X*X-1.D0,
-     1  ASNHCS, NTERMS) )
-      RETURN
- 20   IF (Y.LT.XMAX) DASINH = LOG (Y+SQRT(Y*Y+1.D0))
-      IF (Y.GE.XMAX) DASINH = ALN2 + LOG(Y)
-      DASINH = SIGN (DASINH, X)
-      RETURN
-C
-      END
diff --git a/liboctave/external/slatec-fn/datanh.f b/liboctave/external/slatec-fn/datanh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/datanh.f
+++ /dev/null
@@ -1,89 +0,0 @@
-*DECK DATANH
-      DOUBLE PRECISION FUNCTION DATANH (X)
-C***BEGIN PROLOGUE  DATANH
-C***PURPOSE  Compute the arc hyperbolic tangent.
-C***LIBRARY   SLATEC (FNLIB)
-C***CATEGORY  C4C
-C***TYPE      DOUBLE PRECISION (ATANH-S, DATANH-D, CATANH-C)
-C***KEYWORDS  ARC HYPERBOLIC TANGENT, ATANH, ELEMENTARY FUNCTIONS,
-C             FNLIB, INVERSE HYPERBOLIC TANGENT
-C***AUTHOR  Fullerton, W., (LANL)
-C***DESCRIPTION
-C
-C DATANH(X) calculates the double precision arc hyperbolic
-C tangent for double precision argument X.
-C
-C Series for ATNH       on the interval  0.          to  2.50000E-01
-C                                        with weighted error   6.86E-32
-C                                         log weighted error  31.16
-C                               significant figures required  30.00
-C                                    decimal places required  31.88
-C
-C***REFERENCES  (NONE)
-C***ROUTINES CALLED  D1MACH, DCSEVL, INITDS, XERMSG
-C***REVISION HISTORY  (YYMMDD)
-C   770601  DATE WRITTEN
-C   890531  Changed all specific intrinsics to generic.  (WRB)
-C   890531  REVISION DATE from Version 3.2
-C   891214  Prologue converted to Version 4.0 format.  (BAB)
-C   900315  CALLs to XERROR changed to CALLs to XERMSG.  (THJ)
-C***END PROLOGUE  DATANH
-      DOUBLE PRECISION X, ATNHCS(27), DXREL, SQEPS, Y, DCSEVL, D1MACH
-      LOGICAL FIRST
-      SAVE ATNHCS, NTERMS, DXREL, SQEPS, FIRST
-      DATA ATNHCS(  1) / +.9439510239 3195492308 4289221863 3 D-1      /
-      DATA ATNHCS(  2) / +.4919843705 5786159472 0003457666 8 D-1      /
-      DATA ATNHCS(  3) / +.2102593522 4554327634 7932733175 2 D-2      /
-      DATA ATNHCS(  4) / +.1073554449 7761165846 4073104527 6 D-3      /
-      DATA ATNHCS(  5) / +.5978267249 2930314786 4278751787 2 D-5      /
-      DATA ATNHCS(  6) / +.3505062030 8891348459 6683488620 0 D-6      /
-      DATA ATNHCS(  7) / +.2126374343 7653403508 9621931443 1 D-7      /
-      DATA ATNHCS(  8) / +.1321694535 7155271921 2980172305 5 D-8      /
-      DATA ATNHCS(  9) / +.8365875501 1780703646 2360405295 9 D-10     /
-      DATA ATNHCS( 10) / +.5370503749 3110021638 8143458777 2 D-11     /
-      DATA ATNHCS( 11) / +.3486659470 1571079229 7124578429 0 D-12     /
-      DATA ATNHCS( 12) / +.2284549509 6034330155 2402411972 2 D-13     /
-      DATA ATNHCS( 13) / +.1508407105 9447930448 7422906755 8 D-14     /
-      DATA ATNHCS( 14) / +.1002418816 8041091261 3699572283 7 D-15     /
-      DATA ATNHCS( 15) / +.6698674738 1650695397 1552688298 6 D-17     /
-      DATA ATNHCS( 16) / +.4497954546 4949310830 8332762453 3 D-18     /
-      DATA ATNHCS( 17) / +.3032954474 2794535416 8236714666 6 D-19     /
-      DATA ATNHCS( 18) / +.2052702064 1909368264 6386141866 6 D-20     /
-      DATA ATNHCS( 19) / +.1393848977 0538377131 9301461333 3 D-21     /
-      DATA ATNHCS( 20) / +.9492580637 2245769719 5895466666 6 D-23     /
-      DATA ATNHCS( 21) / +.6481915448 2423076049 8244266666 6 D-24     /
-      DATA ATNHCS( 22) / +.4436730205 7236152726 3232000000 0 D-25     /
-      DATA ATNHCS( 23) / +.3043465618 5431616389 1200000000 0 D-26     /
-      DATA ATNHCS( 24) / +.2091881298 7923934740 4799999999 9 D-27     /
-      DATA ATNHCS( 25) / +.1440445411 2340505613 6533333333 3 D-28     /
-      DATA ATNHCS( 26) / +.9935374683 1416404650 6666666666 6 D-30     /
-      DATA ATNHCS( 27) / +.6863462444 3582600533 3333333333 3 D-31     /
-      DATA FIRST /.TRUE./
-C***FIRST EXECUTABLE STATEMENT  DATANH
-      IF (FIRST) THEN
-         NTERMS = INITDS (ATNHCS, 27, 0.1*REAL(D1MACH(3)) )
-         DXREL = SQRT(D1MACH(4))
-         SQEPS = SQRT(3.0D0*D1MACH(3))
-      ENDIF
-      FIRST = .FALSE.
-C
-      Y = ABS(X)
-      IF (Y .GE. 1.D0) THEN
-         IF (Y .GT. 1.D0) THEN
-            DATANH = (X - X) / (X - X)
-         ELSE
-            DATANH = X / 0.D0
-         ENDIF
-         RETURN
-      ENDIF
-C
-      IF (1.D0-Y .LT. DXREL) CALL XERMSG ('SLATEC', 'DATANH',
-     +   'ANSWER LT HALF PRECISION BECAUSE ABS(X) TOO NEAR 1', 1, 1)
-C
-      DATANH = X
-      IF (Y.GT.SQEPS .AND. Y.LE.0.5D0) DATANH = X*(1.0D0 +
-     1  DCSEVL (8.D0*X*X-1.D0, ATNHCS, NTERMS) )
-      IF (Y.GT.0.5D0) DATANH = 0.5D0*LOG ((1.0D0+X)/(1.0D0-X))
-C
-      RETURN
-      END
diff --git a/liboctave/external/slatec-fn/derf.f b/liboctave/external/slatec-fn/derf.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/derf.f
+++ /dev/null
@@ -1,83 +0,0 @@
-*DECK DERF
-      DOUBLE PRECISION FUNCTION DERF (X)
-C***BEGIN PROLOGUE  DERF
-C***PURPOSE  Compute the error function.
-C***LIBRARY   SLATEC (FNLIB)
-C***CATEGORY  C8A, L5A1E
-C***TYPE      DOUBLE PRECISION (ERF-S, DERF-D)
-C***KEYWORDS  ERF, ERROR FUNCTION, FNLIB, SPECIAL FUNCTIONS
-C***AUTHOR  Fullerton, W., (LANL)
-C***DESCRIPTION
-C
-C DERF(X) calculates the double precision error function for double
-C precision argument X.
-C
-C Series for ERF        on the interval  0.          to  1.00000E+00
-C                                        with weighted error   1.28E-32
-C                                         log weighted error  31.89
-C                               significant figures required  31.05
-C                                    decimal places required  32.55
-C
-C***REFERENCES  (NONE)
-C***ROUTINES CALLED  D1MACH, DCSEVL, DERFC, INITDS
-C***REVISION HISTORY  (YYMMDD)
-C   770701  DATE WRITTEN
-C   890531  Changed all specific intrinsics to generic.  (WRB)
-C   890531  REVISION DATE from Version 3.2
-C   891214  Prologue converted to Version 4.0 format.  (BAB)
-C   900727  Added EXTERNAL statement.  (WRB)
-C   920618  Removed space from variable name.  (RWC, WRB)
-C***END PROLOGUE  DERF
-      DOUBLE PRECISION X, ERFCS(21), SQEPS, SQRTPI, XBIG, Y, D1MACH,
-     1  DCSEVL, DERFC
-      LOGICAL FIRST
-      EXTERNAL DERFC
-      SAVE ERFCS, SQRTPI, NTERF, XBIG, SQEPS, FIRST
-      DATA ERFCS(  1) / -.4904612123 4691808039 9845440333 76 D-1     /
-      DATA ERFCS(  2) / -.1422612051 0371364237 8247418996 31 D+0     /
-      DATA ERFCS(  3) / +.1003558218 7599795575 7546767129 33 D-1     /
-      DATA ERFCS(  4) / -.5768764699 7674847650 8270255091 67 D-3     /
-      DATA ERFCS(  5) / +.2741993125 2196061034 4221607914 71 D-4     /
-      DATA ERFCS(  6) / -.1104317550 7344507604 1353812959 05 D-5     /
-      DATA ERFCS(  7) / +.3848875542 0345036949 9613114981 74 D-7     /
-      DATA ERFCS(  8) / -.1180858253 3875466969 6317518015 81 D-8     /
-      DATA ERFCS(  9) / +.3233421582 6050909646 4029309533 54 D-10    /
-      DATA ERFCS( 10) / -.7991015947 0045487581 6073747085 95 D-12    /
-      DATA ERFCS( 11) / +.1799072511 3961455611 9672454866 34 D-13    /
-      DATA ERFCS( 12) / -.3718635487 8186926382 3168282094 93 D-15    /
-      DATA ERFCS( 13) / +.7103599003 7142529711 6899083946 66 D-17    /
-      DATA ERFCS( 14) / -.1261245511 9155225832 4954248533 33 D-18    /
-      DATA ERFCS( 15) / +.2091640694 1769294369 1705002666 66 D-20    /
-      DATA ERFCS( 16) / -.3253973102 9314072982 3641600000 00 D-22    /
-      DATA ERFCS( 17) / +.4766867209 7976748332 3733333333 33 D-24    /
-      DATA ERFCS( 18) / -.6598012078 2851343155 1999999999 99 D-26    /
-      DATA ERFCS( 19) / +.8655011469 9637626197 3333333333 33 D-28    /
-      DATA ERFCS( 20) / -.1078892517 7498064213 3333333333 33 D-29    /
-      DATA ERFCS( 21) / +.1281188399 3017002666 6666666666 66 D-31    /
-      DATA SQRTPI / 1.772453850 9055160272 9816748334 115D0 /
-      DATA FIRST /.TRUE./
-C***FIRST EXECUTABLE STATEMENT  DERF
-      IF (FIRST) THEN
-         NTERF = INITDS (ERFCS, 21, 0.1*REAL(D1MACH(3)))
-         XBIG = SQRT(-LOG(SQRTPI*D1MACH(3)))
-         SQEPS = SQRT(2.0D0*D1MACH(3))
-      ENDIF
-      FIRST = .FALSE.
-C
-      Y = ABS(X)
-      IF (Y.GT.1.D0) GO TO 20
-C
-C ERF(X) = 1.0 - ERFC(X)  FOR  -1.0 .LE. X .LE. 1.0
-C
-      IF (Y.LE.SQEPS) DERF = 2.0D0*X/SQRTPI
-      IF (Y.GT.SQEPS) DERF = X*(1.0D0 + DCSEVL (2.D0*X*X-1.D0,
-     1  ERFCS, NTERF))
-      RETURN
-C
-C ERF(X) = 1.0 - ERFC(X) FOR ABS(X) .GT. 1.0
-C
- 20   IF (Y.LE.XBIG) DERF = SIGN (1.0D0-DERFC(Y), X)
-      IF (Y.GT.XBIG) DERF = SIGN (1.0D0, X)
-C
-      RETURN
-      END
diff --git a/liboctave/external/slatec-fn/derfc.in.f b/liboctave/external/slatec-fn/derfc.in.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/derfc.in.f
+++ /dev/null
@@ -1,230 +0,0 @@
-*DECK DERFC
-      DOUBLE PRECISION FUNCTION DERFC (X)
-C***BEGIN PROLOGUE  DERFC
-C***PURPOSE  Compute the complementary error function.
-C***LIBRARY   SLATEC (FNLIB)
-C***CATEGORY  C8A, L5A1E
-C***TYPE      DOUBLE PRECISION (ERFC-S, DERFC-D)
-C***KEYWORDS  COMPLEMENTARY ERROR FUNCTION, ERFC, FNLIB,
-C             SPECIAL FUNCTIONS
-C***AUTHOR  Fullerton, W., (LANL)
-C***DESCRIPTION
-C
-C DERFC(X) calculates the double precision complementary error function
-C for double precision argument X.
-C
-C Series for ERF        on the interval  0.          to  1.00000E+00
-C                                        with weighted Error   1.28E-32
-C                                         log weighted Error  31.89
-C                               significant figures required  31.05
-C                                    decimal places required  32.55
-C
-C Series for ERC2       on the interval  2.50000E-01 to  1.00000E+00
-C                                        with weighted Error   2.67E-32
-C                                         log weighted Error  31.57
-C                               significant figures required  30.31
-C                                    decimal places required  32.42
-C
-C Series for ERFC       on the interval  0.          to  2.50000E-01
-C                                        with weighted error   1.53E-31
-C                                         log weighted error  30.82
-C                               significant figures required  29.47
-C                                    decimal places required  31.70
-C
-C***REFERENCES  (NONE)
-C***ROUTINES CALLED  D1MACH, DCSEVL, INITDS, XERMSG
-C***REVISION HISTORY  (YYMMDD)
-C   770701  DATE WRITTEN
-C   890531  Changed all specific intrinsics to generic.  (WRB)
-C   890531  REVISION DATE from Version 3.2
-C   891214  Prologue converted to Version 4.0 format.  (BAB)
-C   900315  CALLs to XERROR changed to CALLs to XERMSG.  (THJ)
-C   920618  Removed space from variable names.  (RWC, WRB)
-C***END PROLOGUE  DERFC
-      DOUBLE PRECISION X, ERFCS(21), ERFCCS(59), ERC2CS(49), SQEPS,
-     1  SQRTPI, XMAX, TXMAX, XSML, Y, D1MACH, DCSEVL
-      LOGICAL FIRST
-      SAVE ERFCS, ERC2CS, ERFCCS, SQRTPI, NTERF,
-     1 NTERFC, NTERC2, XSML, XMAX, SQEPS, FIRST
-      DATA ERFCS(  1) / -.4904612123 4691808039 9845440333 76 D-1     /
-      DATA ERFCS(  2) / -.1422612051 0371364237 8247418996 31 D+0     /
-      DATA ERFCS(  3) / +.1003558218 7599795575 7546767129 33 D-1     /
-      DATA ERFCS(  4) / -.5768764699 7674847650 8270255091 67 D-3     /
-      DATA ERFCS(  5) / +.2741993125 2196061034 4221607914 71 D-4     /
-      DATA ERFCS(  6) / -.1104317550 7344507604 1353812959 05 D-5     /
-      DATA ERFCS(  7) / +.3848875542 0345036949 9613114981 74 D-7     /
-      DATA ERFCS(  8) / -.1180858253 3875466969 6317518015 81 D-8     /
-      DATA ERFCS(  9) / +.3233421582 6050909646 4029309533 54 D-10    /
-      DATA ERFCS( 10) / -.7991015947 0045487581 6073747085 95 D-12    /
-      DATA ERFCS( 11) / +.1799072511 3961455611 9672454866 34 D-13    /
-      DATA ERFCS( 12) / -.3718635487 8186926382 3168282094 93 D-15    /
-      DATA ERFCS( 13) / +.7103599003 7142529711 6899083946 66 D-17    /
-      DATA ERFCS( 14) / -.1261245511 9155225832 4954248533 33 D-18    /
-      DATA ERFCS( 15) / +.2091640694 1769294369 1705002666 66 D-20    /
-      DATA ERFCS( 16) / -.3253973102 9314072982 3641600000 00 D-22    /
-      DATA ERFCS( 17) / +.4766867209 7976748332 3733333333 33 D-24    /
-      DATA ERFCS( 18) / -.6598012078 2851343155 1999999999 99 D-26    /
-      DATA ERFCS( 19) / +.8655011469 9637626197 3333333333 33 D-28    /
-      DATA ERFCS( 20) / -.1078892517 7498064213 3333333333 33 D-29    /
-      DATA ERFCS( 21) / +.1281188399 3017002666 6666666666 66 D-31    /
-      DATA ERC2CS(  1) / -.6960134660 2309501127 3915082619 7 D-1      /
-      DATA ERC2CS(  2) / -.4110133936 2620893489 8221208466 6 D-1      /
-      DATA ERC2CS(  3) / +.3914495866 6896268815 6114370524 4 D-2      /
-      DATA ERC2CS(  4) / -.4906395650 5489791612 8093545077 4 D-3      /
-      DATA ERC2CS(  5) / +.7157479001 3770363807 6089414182 5 D-4      /
-      DATA ERC2CS(  6) / -.1153071634 1312328338 0823284791 2 D-4      /
-      DATA ERC2CS(  7) / +.1994670590 2019976350 5231486770 9 D-5      /
-      DATA ERC2CS(  8) / -.3642666471 5992228739 3611843071 1 D-6      /
-      DATA ERC2CS(  9) / +.6944372610 0050125899 3127721463 3 D-7      /
-      DATA ERC2CS( 10) / -.1371220902 1043660195 3460514121 0 D-7      /
-      DATA ERC2CS( 11) / +.2788389661 0071371319 6386034808 7 D-8      /
-      DATA ERC2CS( 12) / -.5814164724 3311615518 6479105031 6 D-9      /
-      DATA ERC2CS( 13) / +.1238920491 7527531811 8016881795 0 D-9      /
-      DATA ERC2CS( 14) / -.2690639145 3067434323 9042493788 9 D-10     /
-      DATA ERC2CS( 15) / +.5942614350 8479109824 4470968384 0 D-11     /
-      DATA ERC2CS( 16) / -.1332386735 7581195792 8775442057 0 D-11     /
-      DATA ERC2CS( 17) / +.3028046806 1771320171 7369724330 4 D-12     /
-      DATA ERC2CS( 18) / -.6966648814 9410325887 9586758895 4 D-13     /
-      DATA ERC2CS( 19) / +.1620854541 0539229698 1289322762 8 D-13     /
-      DATA ERC2CS( 20) / -.3809934465 2504919998 7691305772 9 D-14     /
-      DATA ERC2CS( 21) / +.9040487815 9788311493 6897101297 5 D-15     /
-      DATA ERC2CS( 22) / -.2164006195 0896073478 0981204700 3 D-15     /
-      DATA ERC2CS( 23) / +.5222102233 9958549846 0798024417 2 D-16     /
-      DATA ERC2CS( 24) / -.1269729602 3645553363 7241552778 0 D-16     /
-      DATA ERC2CS( 25) / +.3109145504 2761975838 3622741295 1 D-17     /
-      DATA ERC2CS( 26) / -.7663762920 3203855240 0956671481 1 D-18     /
-      DATA ERC2CS( 27) / +.1900819251 3627452025 3692973329 0 D-18     /
-      DATA ERC2CS( 28) / -.4742207279 0690395452 2565599996 5 D-19     /
-      DATA ERC2CS( 29) / +.1189649200 0765283828 8068307845 1 D-19     /
-      DATA ERC2CS( 30) / -.3000035590 3257802568 4527131306 6 D-20     /
-      DATA ERC2CS( 31) / +.7602993453 0432461730 1938527709 8 D-21     /
-      DATA ERC2CS( 32) / -.1935909447 6068728815 6981104913 0 D-21     /
-      DATA ERC2CS( 33) / +.4951399124 7733378810 0004238677 3 D-22     /
-      DATA ERC2CS( 34) / -.1271807481 3363718796 0862198988 8 D-22     /
-      DATA ERC2CS( 35) / +.3280049600 4695130433 1584165205 3 D-23     /
-      DATA ERC2CS( 36) / -.8492320176 8228965689 2479242239 9 D-24     /
-      DATA ERC2CS( 37) / +.2206917892 8075602235 1987998719 9 D-24     /
-      DATA ERC2CS( 38) / -.5755617245 6965284983 1281950719 9 D-25     /
-      DATA ERC2CS( 39) / +.1506191533 6392342503 5414405119 9 D-25     /
-      DATA ERC2CS( 40) / -.3954502959 0187969531 0428569599 9 D-26     /
-      DATA ERC2CS( 41) / +.1041529704 1515009799 8464505173 3 D-26     /
-      DATA ERC2CS( 42) / -.2751487795 2787650794 5017890133 3 D-27     /
-      DATA ERC2CS( 43) / +.7290058205 4975574089 9770368000 0 D-28     /
-      DATA ERC2CS( 44) / -.1936939645 9159478040 7750109866 6 D-28     /
-      DATA ERC2CS( 45) / +.5160357112 0514872983 7005482666 6 D-29     /
-      DATA ERC2CS( 46) / -.1378419322 1930940993 8964480000 0 D-29     /
-      DATA ERC2CS( 47) / +.3691326793 1070690422 5109333333 3 D-30     /
-      DATA ERC2CS( 48) / -.9909389590 6243654206 5322666666 6 D-31     /
-      DATA ERC2CS( 49) / +.2666491705 1953884133 2394666666 6 D-31     /
-      DATA ERFCCS(  1) / +.7151793102 0292477450 3697709496 D-1        /
-      DATA ERFCCS(  2) / -.2653243433 7606715755 8893386681 D-1        /
-      DATA ERFCCS(  3) / +.1711153977 9208558833 2699194606 D-2        /
-      DATA ERFCCS(  4) / -.1637516634 5851788416 3746404749 D-3        /
-      DATA ERFCCS(  5) / +.1987129350 0552036499 5974806758 D-4        /
-      DATA ERFCCS(  6) / -.2843712412 7665550875 0175183152 D-5        /
-      DATA ERFCCS(  7) / +.4606161308 9631303696 9379968464 D-6        /
-      DATA ERFCCS(  8) / -.8227753025 8792084205 7766536366 D-7        /
-      DATA ERFCCS(  9) / +.1592141872 7709011298 9358340826 D-7        /
-      DATA ERFCCS( 10) / -.3295071362 2528432148 6631665072 D-8        /
-      DATA ERFCCS( 11) / +.7223439760 4005554658 1261153890 D-9        /
-      DATA ERFCCS( 12) / -.1664855813 3987295934 4695966886 D-9        /
-      DATA ERFCCS( 13) / +.4010392588 2376648207 7671768814 D-10       /
-      DATA ERFCCS( 14) / -.1004816214 4257311327 2170176283 D-10       /
-      DATA ERFCCS( 15) / +.2608275913 3003338085 9341009439 D-11       /
-      DATA ERFCCS( 16) / -.6991110560 4040248655 7697812476 D-12       /
-      DATA ERFCCS( 17) / +.1929492333 2617070862 4205749803 D-12       /
-      DATA ERFCCS( 18) / -.5470131188 7543310649 0125085271 D-13       /
-      DATA ERFCCS( 19) / +.1589663309 7626974483 9084032762 D-13       /
-      DATA ERFCCS( 20) / -.4726893980 1975548392 0369584290 D-14       /
-      DATA ERFCCS( 21) / +.1435873376 7849847867 2873997840 D-14       /
-      DATA ERFCCS( 22) / -.4449510561 8173583941 7250062829 D-15       /
-      DATA ERFCCS( 23) / +.1404810884 7682334373 7305537466 D-15       /
-      DATA ERFCCS( 24) / -.4513818387 7642108962 5963281623 D-16       /
-      DATA ERFCCS( 25) / +.1474521541 0451330778 7018713262 D-16       /
-      DATA ERFCCS( 26) / -.4892621406 9457761543 6841552532 D-17       /
-      DATA ERFCCS( 27) / +.1647612141 4106467389 5301522827 D-17       /
-      DATA ERFCCS( 28) / -.5626817176 3294080929 9928521323 D-18       /
-      DATA ERFCCS( 29) / +.1947443382 2320785142 9197867821 D-18       /
-      DATA ERFCCS( 30) / -.6826305642 9484207295 6664144723 D-19       /
-      DATA ERFCCS( 31) / +.2421988887 2986492401 8301125438 D-19       /
-      DATA ERFCCS( 32) / -.8693414133 5030704256 3800861857 D-20       /
-      DATA ERFCCS( 33) / +.3155180346 2280855712 2363401262 D-20       /
-      DATA ERFCCS( 34) / -.1157372324 0496087426 1239486742 D-20       /
-      DATA ERFCCS( 35) / +.4288947161 6056539462 3737097442 D-21       /
-      DATA ERFCCS( 36) / -.1605030742 0576168500 5737770964 D-21       /
-      DATA ERFCCS( 37) / +.6063298757 4538026449 5069923027 D-22       /
-      DATA ERFCCS( 38) / -.2311404251 6979584909 8840801367 D-22       /
-      DATA ERFCCS( 39) / +.8888778540 6618855255 4702955697 D-23       /
-      DATA ERFCCS( 40) / -.3447260576 6513765223 0718495566 D-23       /
-      DATA ERFCCS( 41) / +.1347865460 2069650682 7582774181 D-23       /
-      DATA ERFCCS( 42) / -.5311794071 1250217364 5873201807 D-24       /
-      DATA ERFCCS( 43) / +.2109341058 6197831682 8954734537 D-24       /
-      DATA ERFCCS( 44) / -.8438365587 9237891159 8133256738 D-25       /
-      DATA ERFCCS( 45) / +.3399982524 9452089062 7359576337 D-25       /
-      DATA ERFCCS( 46) / -.1379452388 0732420900 2238377110 D-25       /
-      DATA ERFCCS( 47) / +.5634490311 8332526151 3392634811 D-26       /
-      DATA ERFCCS( 48) / -.2316490434 4770654482 3427752700 D-26       /
-      DATA ERFCCS( 49) / +.9584462844 6018101526 3158381226 D-27       /
-      DATA ERFCCS( 50) / -.3990722880 3301097262 4224850193 D-27       /
-      DATA ERFCCS( 51) / +.1672129225 9444773601 7228709669 D-27       /
-      DATA ERFCCS( 52) / -.7045991522 7660138563 8803782587 D-28       /
-      DATA ERFCCS( 53) / +.2979768402 8642063541 2357989444 D-28       /
-      DATA ERFCCS( 54) / -.1262522466 4606192972 2422632994 D-28       /
-      DATA ERFCCS( 55) / +.5395438704 5424879398 5299653154 D-29       /
-      DATA ERFCCS( 56) / -.2380992882 5314591867 5346190062 D-29       /
-      DATA ERFCCS( 57) / +.1099052830 1027615735 9726683750 D-29       /
-      DATA ERFCCS( 58) / -.4867713741 6449657273 2518677435 D-30       /
-      DATA ERFCCS( 59) / +.1525877264 1103575676 3200828211 D-30       /
-      DATA SQRTPI / 1.772453850 9055160272 9816748334 115D0 /
-      DATA FIRST /.TRUE./
-C***FIRST EXECUTABLE STATEMENT  DERFC
-      IF (FIRST) THEN
-         ETA = 0.1*REAL(D1MACH(3))
-         NTERF = INITDS (ERFCS, 21, ETA)
-         NTERFC = INITDS (ERFCCS, 59, ETA)
-         NTERC2 = INITDS (ERC2CS, 49, ETA)
-C
-         XSML = -SQRT(-LOG(SQRTPI*D1MACH(3)))
-         TXMAX = SQRT(-LOG(SQRTPI*D1MACH(1)))
-         XMAX = TXMAX - 0.5D0*LOG(TXMAX)/TXMAX - 0.01D0
-         SQEPS = SQRT(2.0D0*D1MACH(3))
-      ENDIF
-      FIRST = .FALSE.
-C
-      IF (ISNAN(X)) THEN
-         DERFC = X
-         RETURN
-      ENDIF
-C
-      IF (X.GT.XSML) GO TO 20
-C
-C ERFC(X) = 1.0 - ERF(X)  FOR  X .LT. XSML
-C
-      DERFC = 2.0D0
-      RETURN
-C
- 20   IF (X.GT.XMAX) GO TO 40
-      Y = ABS(X)
-      IF (Y.GT.1.0D0) GO TO 30
-C
-C ERFC(X) = 1.0 - ERF(X)  FOR ABS(X) .LE. 1.0
-C
-      IF (Y.LT.SQEPS) DERFC = 1.0D0 - 2.0D0*X/SQRTPI
-      IF (Y.GE.SQEPS) DERFC = 1.0D0 - X*(1.0D0 + DCSEVL (2.D0*X*X-1.D0,
-     1  ERFCS, NTERF))
-      RETURN
-C
-C ERFC(X) = 1.0 - ERF(X)  FOR  1.0 .LT. ABS(X) .LE. XMAX
-C
- 30   Y = Y*Y
-      IF (Y.LE.4.D0) DERFC = EXP(-Y)/ABS(X) * (0.5D0 + DCSEVL (
-     1  (8.D0/Y-5.D0)/3.D0, ERC2CS, NTERC2) )
-      IF (Y.GT.4.D0) DERFC = EXP(-Y)/ABS(X) * (0.5D0 + DCSEVL (
-     1  8.D0/Y-1.D0, ERFCCS, NTERFC) )
-      IF (X.LT.0.D0) DERFC = 2.0D0 - DERFC
-      RETURN
-C
- 40   DERFC = 0.D0
-      RETURN
-C
-      END
diff --git a/liboctave/external/slatec-fn/erf.f b/liboctave/external/slatec-fn/erf.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/erf.f
+++ /dev/null
@@ -1,73 +0,0 @@
-*DECK ERF
-      FUNCTION ERF (X)
-C***BEGIN PROLOGUE  ERF
-C***PURPOSE  Compute the error function.
-C***LIBRARY   SLATEC (FNLIB)
-C***CATEGORY  C8A, L5A1E
-C***TYPE      SINGLE PRECISION (ERF-S, DERF-D)
-C***KEYWORDS  ERF, ERROR FUNCTION, FNLIB, SPECIAL FUNCTIONS
-C***AUTHOR  Fullerton, W., (LANL)
-C***DESCRIPTION
-C
-C ERF(X) calculates the single precision error function for
-C single precision argument X.
-C
-C Series for ERF        on the interval  0.          to  1.00000D+00
-C                                        with weighted error   7.10E-18
-C                                         log weighted error  17.15
-C                               significant figures required  16.31
-C                                    decimal places required  17.71
-C
-C***REFERENCES  (NONE)
-C***ROUTINES CALLED  CSEVL, ERFC, INITS, R1MACH
-C***REVISION HISTORY  (YYMMDD)
-C   770401  DATE WRITTEN
-C   890531  Changed all specific intrinsics to generic.  (WRB)
-C   890531  REVISION DATE from Version 3.2
-C   891214  Prologue converted to Version 4.0 format.  (BAB)
-C   900727  Added EXTERNAL statement.  (WRB)
-C   920618  Removed space from variable name.  (RWC, WRB)
-C***END PROLOGUE  ERF
-      DIMENSION ERFCS(13)
-      LOGICAL FIRST
-      EXTERNAL ERFC
-      SAVE ERFCS, SQRTPI, NTERF, XBIG, SQEPS, FIRST
-      DATA ERFCS( 1) /   -.0490461212 34691808E0 /
-      DATA ERFCS( 2) /   -.1422612051 0371364E0 /
-      DATA ERFCS( 3) /    .0100355821 87599796E0 /
-      DATA ERFCS( 4) /   -.0005768764 69976748E0 /
-      DATA ERFCS( 5) /    .0000274199 31252196E0 /
-      DATA ERFCS( 6) /   -.0000011043 17550734E0 /
-      DATA ERFCS( 7) /    .0000000384 88755420E0 /
-      DATA ERFCS( 8) /   -.0000000011 80858253E0 /
-      DATA ERFCS( 9) /    .0000000000 32334215E0 /
-      DATA ERFCS(10) /   -.0000000000 00799101E0 /
-      DATA ERFCS(11) /    .0000000000 00017990E0 /
-      DATA ERFCS(12) /   -.0000000000 00000371E0 /
-      DATA ERFCS(13) /    .0000000000 00000007E0 /
-      DATA SQRTPI /1.772453850 9055160E0/
-      DATA FIRST /.TRUE./
-C***FIRST EXECUTABLE STATEMENT  ERF
-      IF (FIRST) THEN
-         NTERF = INITS (ERFCS, 13, 0.1*R1MACH(3))
-         XBIG = SQRT(-LOG(SQRTPI*R1MACH(3)))
-         SQEPS = SQRT(2.0*R1MACH(3))
-      ENDIF
-      FIRST = .FALSE.
-C
-      Y = ABS(X)
-      IF (Y.GT.1.) GO TO 20
-C
-C ERF(X) = 1. - ERFC(X) FOR -1. .LE. X .LE. 1.
-C
-      IF (Y.LE.SQEPS) ERF = 2.0*X/SQRTPI
-      IF (Y.GT.SQEPS) ERF = X*(1.0 + CSEVL(2.*X**2-1., ERFCS, NTERF))
-      RETURN
-C
-C ERF(X) = 1. - ERFC(X) FOR  ABS(X) .GT. 1.
-C
- 20   IF (Y.LE.XBIG) ERF = SIGN (1.0-ERFC(Y), X)
-      IF (Y.GT.XBIG) ERF = SIGN (1.0, X)
-C
-      RETURN
-      END
diff --git a/liboctave/external/slatec-fn/erfc.in.f b/liboctave/external/slatec-fn/erfc.in.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/erfc.in.f
+++ /dev/null
@@ -1,160 +0,0 @@
-*DECK ERFC
-      FUNCTION ERFC (X)
-C***BEGIN PROLOGUE  ERFC
-C***PURPOSE  Compute the complementary error function.
-C***LIBRARY   SLATEC (FNLIB)
-C***CATEGORY  C8A, L5A1E
-C***TYPE      SINGLE PRECISION (ERFC-S, DERFC-D)
-C***KEYWORDS  COMPLEMENTARY ERROR FUNCTION, ERFC, FNLIB,
-C             SPECIAL FUNCTIONS
-C***AUTHOR  Fullerton, W., (LANL)
-C***DESCRIPTION
-C
-C ERFC(X) calculates the single precision complementary error
-C function for single precision argument X.
-C
-C Series for ERF        on the interval  0.          to  1.00000D+00
-C                                        with weighted error   7.10E-18
-C                                         log weighted error  17.15
-C                               significant figures required  16.31
-C                                    decimal places required  17.71
-C
-C Series for ERFC       on the interval  0.          to  2.50000D-01
-C                                        with weighted error   4.81E-17
-C                                         log weighted error  16.32
-C                        approx significant figures required  15.0
-C
-C
-C Series for ERC2       on the interval  2.50000D-01 to  1.00000D+00
-C                                        with weighted error   5.22E-17
-C                                         log weighted error  16.28
-C                        approx significant figures required  15.0
-C                                    decimal places required  16.96
-C
-C***REFERENCES  (NONE)
-C***ROUTINES CALLED  CSEVL, INITS, R1MACH, XERMSG
-C***REVISION HISTORY  (YYMMDD)
-C   770701  DATE WRITTEN
-C   890531  Changed all specific intrinsics to generic.  (WRB)
-C   890531  REVISION DATE from Version 3.2
-C   891214  Prologue converted to Version 4.0 format.  (BAB)
-C   900315  CALLs to XERROR changed to CALLs to XERMSG.  (THJ)
-C   920618  Removed space from variable names.  (RWC, WRB)
-C***END PROLOGUE  ERFC
-      DIMENSION ERFCS(13), ERFCCS(24), ERC2CS(23)
-      LOGICAL FIRST
-      SAVE ERFCS, ERC2CS, ERFCCS, SQRTPI, NTERF, NTERFC,
-     1 NTERC2, XSML, XMAX, SQEPS, FIRST
-      DATA ERFCS( 1) /   -.0490461212 34691808E0 /
-      DATA ERFCS( 2) /   -.1422612051 0371364E0 /
-      DATA ERFCS( 3) /    .0100355821 87599796E0 /
-      DATA ERFCS( 4) /   -.0005768764 69976748E0 /
-      DATA ERFCS( 5) /    .0000274199 31252196E0 /
-      DATA ERFCS( 6) /   -.0000011043 17550734E0 /
-      DATA ERFCS( 7) /    .0000000384 88755420E0 /
-      DATA ERFCS( 8) /   -.0000000011 80858253E0 /
-      DATA ERFCS( 9) /    .0000000000 32334215E0 /
-      DATA ERFCS(10) /   -.0000000000 00799101E0 /
-      DATA ERFCS(11) /    .0000000000 00017990E0 /
-      DATA ERFCS(12) /   -.0000000000 00000371E0 /
-      DATA ERFCS(13) /    .0000000000 00000007E0 /
-      DATA ERC2CS( 1) /   -.0696013466 02309501E0 /
-      DATA ERC2CS( 2) /   -.0411013393 62620893E0 /
-      DATA ERC2CS( 3) /    .0039144958 66689626E0 /
-      DATA ERC2CS( 4) /   -.0004906395 65054897E0 /
-      DATA ERC2CS( 5) /    .0000715747 90013770E0 /
-      DATA ERC2CS( 6) /   -.0000115307 16341312E0 /
-      DATA ERC2CS( 7) /    .0000019946 70590201E0 /
-      DATA ERC2CS( 8) /   -.0000003642 66647159E0 /
-      DATA ERC2CS( 9) /    .0000000694 43726100E0 /
-      DATA ERC2CS(10) /   -.0000000137 12209021E0 /
-      DATA ERC2CS(11) /    .0000000027 88389661E0 /
-      DATA ERC2CS(12) /   -.0000000005 81416472E0 /
-      DATA ERC2CS(13) /    .0000000001 23892049E0 /
-      DATA ERC2CS(14) /   -.0000000000 26906391E0 /
-      DATA ERC2CS(15) /    .0000000000 05942614E0 /
-      DATA ERC2CS(16) /   -.0000000000 01332386E0 /
-      DATA ERC2CS(17) /    .0000000000 00302804E0 /
-      DATA ERC2CS(18) /   -.0000000000 00069666E0 /
-      DATA ERC2CS(19) /    .0000000000 00016208E0 /
-      DATA ERC2CS(20) /   -.0000000000 00003809E0 /
-      DATA ERC2CS(21) /    .0000000000 00000904E0 /
-      DATA ERC2CS(22) /   -.0000000000 00000216E0 /
-      DATA ERC2CS(23) /    .0000000000 00000052E0 /
-      DATA ERFCCS( 1) /   0.0715179310 202925E0 /
-      DATA ERFCCS( 2) /   -.0265324343 37606719E0 /
-      DATA ERFCCS( 3) /    .0017111539 77920853E0 /
-      DATA ERFCCS( 4) /   -.0001637516 63458512E0 /
-      DATA ERFCCS( 5) /    .0000198712 93500549E0 /
-      DATA ERFCCS( 6) /   -.0000028437 12412769E0 /
-      DATA ERFCCS( 7) /    .0000004606 16130901E0 /
-      DATA ERFCCS( 8) /   -.0000000822 77530261E0 /
-      DATA ERFCCS( 9) /    .0000000159 21418724E0 /
-      DATA ERFCCS(10) /   -.0000000032 95071356E0 /
-      DATA ERFCCS(11) /    .0000000007 22343973E0 /
-      DATA ERFCCS(12) /   -.0000000001 66485584E0 /
-      DATA ERFCCS(13) /    .0000000000 40103931E0 /
-      DATA ERFCCS(14) /   -.0000000000 10048164E0 /
-      DATA ERFCCS(15) /    .0000000000 02608272E0 /
-      DATA ERFCCS(16) /   -.0000000000 00699105E0 /
-      DATA ERFCCS(17) /    .0000000000 00192946E0 /
-      DATA ERFCCS(18) /   -.0000000000 00054704E0 /
-      DATA ERFCCS(19) /    .0000000000 00015901E0 /
-      DATA ERFCCS(20) /   -.0000000000 00004729E0 /
-      DATA ERFCCS(21) /    .0000000000 00001432E0 /
-      DATA ERFCCS(22) /   -.0000000000 00000439E0 /
-      DATA ERFCCS(23) /    .0000000000 00000138E0 /
-      DATA ERFCCS(24) /   -.0000000000 00000048E0 /
-      DATA SQRTPI /1.772453850 9055160E0/
-      DATA FIRST /.TRUE./
-C***FIRST EXECUTABLE STATEMENT  ERFC
-      IF (FIRST) THEN
-         ETA = 0.1*R1MACH(3)
-         NTERF = INITS (ERFCS, 13, ETA)
-         NTERFC = INITS (ERFCCS, 24, ETA)
-         NTERC2 = INITS (ERC2CS, 23, ETA)
-C
-         XSML = -SQRT (-LOG(SQRTPI*R1MACH(3)))
-         TXMAX = SQRT (-LOG(SQRTPI*R1MACH(1)))
-         XMAX = TXMAX - 0.5*LOG(TXMAX)/TXMAX - 0.01
-         SQEPS = SQRT (2.0*R1MACH(3))
-      ENDIF
-      FIRST = .FALSE.
-C
-      IF (ISNAN(X)) THEN
-         ERFC = X
-         RETURN
-      ENDIF
-C
-      IF (X.GT.XSML) GO TO 20
-C
-C ERFC(X) = 1.0 - ERF(X) FOR X .LT. XSML
-C
-      ERFC = 2.
-      RETURN
-C
- 20   IF (X.GT.XMAX) GO TO 40
-      Y = ABS(X)
-      IF (Y.GT.1.0) GO TO 30
-C
-C ERFC(X) = 1.0 - ERF(X) FOR -1. .LE. X .LE. 1.
-C
-      IF (Y.LT.SQEPS) ERFC = 1.0 - 2.0*X/SQRTPI
-      IF (Y.GE.SQEPS) ERFC = 1.0 -
-     1  X*(1.0 + CSEVL (2.*X*X-1., ERFCS, NTERF) )
-      RETURN
-C
-C ERFC(X) = 1.0 - ERF(X) FOR 1. .LT. ABS(X) .LE. XMAX
-C
- 30   Y = Y*Y
-      IF (Y.LE.4.) ERFC = EXP(-Y)/ABS(X) * (0.5 + CSEVL ((8./Y-5.)/3.,
-     1  ERC2CS, NTERC2) )
-      IF (Y.GT.4.) ERFC = EXP(-Y)/ABS(X) * (0.5 + CSEVL (8./Y-1.,
-     1  ERFCCS, NTERFC) )
-      IF (X.LT.0.) ERFC = 2.0 - ERFC
-      RETURN
-C
- 40   ERFC = 0.
-      RETURN
-C
-      END
diff --git a/liboctave/external/slatec-fn/module.mk b/liboctave/external/slatec-fn/module.mk
--- a/liboctave/external/slatec-fn/module.mk
+++ b/liboctave/external/slatec-fn/module.mk
@@ -1,81 +1,53 @@
 EXTERNAL_SOURCES += \
   %reldir%/albeta.f \
   %reldir%/alngam.f \
   %reldir%/alnrel.f \
   %reldir%/algams.f \
-  %reldir%/acosh.f \
-  %reldir%/asinh.f \
-  %reldir%/atanh.f \
   %reldir%/betai.f \
   %reldir%/csevl.f \
   %reldir%/d9gmit.f \
   %reldir%/d9lgic.f \
   %reldir%/d9lgit.f \
   %reldir%/d9lgmc.f \
-  %reldir%/dacosh.f \
-  %reldir%/dasinh.f \
-  %reldir%/datanh.f \
   %reldir%/dbetai.f \
   %reldir%/dcsevl.f \
-  %reldir%/derf.f \
   %reldir%/dgami.f \
   %reldir%/dgamit.f \
   %reldir%/dgamlm.f \
   %reldir%/dgamma.f \
   %reldir%/dgamr.f \
   %reldir%/dlbeta.f \
   %reldir%/dlgams.f \
   %reldir%/dlngam.f \
   %reldir%/dlnrel.f \
   %reldir%/dpchim.f \
   %reldir%/dpchst.f \
   %reldir%/dpsifn.f \
-  %reldir%/erf.f \
   %reldir%/gami.f \
   %reldir%/gamit.f \
   %reldir%/gamlim.f \
   %reldir%/gamma.f \
   %reldir%/gamr.f \
   %reldir%/initds.f \
   %reldir%/inits.f \
   %reldir%/pchim.f \
   %reldir%/pchst.f \
   %reldir%/psifn.f \
   %reldir%/r9lgmc.f \
   %reldir%/r9lgit.f \
   %reldir%/r9gmit.f \
   %reldir%/r9lgic.f \
-  %reldir%/xdacosh.f \
-  %reldir%/xdasinh.f \
-  %reldir%/xdatanh.f \
   %reldir%/xdbetai.f \
-  %reldir%/xderf.f \
-  %reldir%/xderfc.f \
   %reldir%/xdgami.f \
   %reldir%/xdgamit.f \
   %reldir%/xdgamma.f \
   %reldir%/xgmainc.f \
-  %reldir%/xacosh.f \
-  %reldir%/xasinh.f \
-  %reldir%/xatanh.f \
-  %reldir%/xerf.f \
-  %reldir%/xerfc.f \
   %reldir%/xsgmainc.f \
   %reldir%/xgamma.f \
   %reldir%/xbetai.f
 
-nodist_liboctave_external_libexternal_la_SOURCES += \
-  %reldir%/derfc.f \
-  %reldir%/erfc.f
-
-%reldir%/erfc.f: %reldir%/erfc.in.f build-aux/subst-f77-isnan-macro.sh | %reldir%/$(octave_dirstamp)
-	$(AM_V_GEN)$(call simple-filter-rule,build-aux/subst-f77-isnan-macro.sh)
-
-%reldir%/derfc.f: %reldir%/derfc.in.f build-aux/subst-f77-isnan-macro.sh | %reldir%/$(octave_dirstamp)
-	$(AM_V_GEN)$(call simple-filter-rule,build-aux/subst-f77-isnan-macro.sh)
-
 liboctave_EXTRA_DIST += \
   %reldir%/derfc.in.f \
   %reldir%/erfc.in.f
 
 DIRSTAMP_FILES += %reldir%/$(octave_dirstamp)
diff --git a/liboctave/external/slatec-fn/xacosh.f b/liboctave/external/slatec-fn/xacosh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/xacosh.f
+++ /dev/null
@@ -1,6 +0,0 @@
-      subroutine xacosh (x, result)
-      external acosh
-      real x, result, acosh
-      result = acosh (x)
-      return
-      end
diff --git a/liboctave/external/slatec-fn/xasinh.f b/liboctave/external/slatec-fn/xasinh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/xasinh.f
+++ /dev/null
@@ -1,6 +0,0 @@
-      subroutine xasinh (x, result)
-      external asinh
-      real x, result, asinh
-      result = asinh (x)
-      return
-      end
diff --git a/liboctave/external/slatec-fn/xatanh.f b/liboctave/external/slatec-fn/xatanh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/xatanh.f
+++ /dev/null
@@ -1,6 +0,0 @@
-      subroutine xatanh (x, result)
-      external atanh
-      real x, result, atanh
-      result = atanh (x)
-      return
-      end
diff --git a/liboctave/external/slatec-fn/xdacosh.f b/liboctave/external/slatec-fn/xdacosh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/xdacosh.f
+++ /dev/null
@@ -1,6 +0,0 @@
-      subroutine xdacosh (x, result)
-      external dacosh
-      double precision x, result, dacosh
-      result = dacosh (x)
-      return
-      end
diff --git a/liboctave/external/slatec-fn/xdasinh.f b/liboctave/external/slatec-fn/xdasinh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/xdasinh.f
+++ /dev/null
@@ -1,6 +0,0 @@
-      subroutine xdasinh (x, result)
-      external dasinh
-      double precision x, result, dasinh
-      result = dasinh (x)
-      return
-      end
diff --git a/liboctave/external/slatec-fn/xdatanh.f b/liboctave/external/slatec-fn/xdatanh.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/xdatanh.f
+++ /dev/null
@@ -1,6 +0,0 @@
-      subroutine xdatanh (x, result)
-      external datanh
-      double precision x, result, datanh
-      result = datanh (x)
-      return
-      end
diff --git a/liboctave/external/slatec-fn/xderf.f b/liboctave/external/slatec-fn/xderf.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/xderf.f
+++ /dev/null
@@ -1,6 +0,0 @@
-      subroutine xderf (x, result)
-      external derf
-      double precision x, result, derf
-      result = derf (x)
-      return
-      end
diff --git a/liboctave/external/slatec-fn/xderfc.f b/liboctave/external/slatec-fn/xderfc.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/xderfc.f
+++ /dev/null
@@ -1,6 +0,0 @@
-      subroutine xderfc (x, result)
-      external derfc
-      double precision x, result, derfc
-      result = derfc (x)
-      return
-      end
diff --git a/liboctave/external/slatec-fn/xerf.f b/liboctave/external/slatec-fn/xerf.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/xerf.f
+++ /dev/null
@@ -1,6 +0,0 @@
-      subroutine xerf (x, result)
-      external erf
-      real x, result, erf
-      result = erf (x)
-      return
-      end
diff --git a/liboctave/external/slatec-fn/xerfc.f b/liboctave/external/slatec-fn/xerfc.f
deleted file mode 100644
--- a/liboctave/external/slatec-fn/xerfc.f
+++ /dev/null
@@ -1,6 +0,0 @@
-      subroutine xerfc (x, result)
-      external erfc
-      real x, result, erfc
-      result = erfc (x)
-      return
-      end
diff --git a/liboctave/numeric/lo-slatec-proto.h b/liboctave/numeric/lo-slatec-proto.h
--- a/liboctave/numeric/lo-slatec-proto.h
+++ b/liboctave/numeric/lo-slatec-proto.h
@@ -24,92 +24,36 @@ along with Octave; see the file COPYING.
 #define octave_lo_slatec_proto_h 1
 
 #include "octave-config.h"
 
 #include "f77-fcn.h"
 
 extern "C"
 {
-  // ACOSH
-
-  F77_RET_T
-  F77_FUNC (xacosh, XACOSH) (const F77_REAL&, F77_REAL&);
-
-  F77_RET_T
-  F77_FUNC (xdacosh, XDACOSH) (const F77_DBLE&, F77_DBLE&);
-
-  // ASINH
-
-  F77_RET_T
-  F77_FUNC (xasinh, XASINH) (const F77_REAL&, F77_REAL&);
-
-  F77_RET_T
-  F77_FUNC (xdasinh, XDASINH) (const F77_DBLE&, F77_DBLE&);
-
-  // ATANH
-
-  F77_RET_T
-  F77_FUNC (xatanh, XATANH) (const F77_REAL&, F77_REAL&);
-
-  F77_RET_T
-  F77_FUNC (xdatanh, XDATANH) (const F77_DBLE&, F77_DBLE&);
-
   // BETAI
 
   F77_RET_T
   F77_FUNC (xbetai, XBETAI) (const F77_REAL&, const F77_REAL&,
                              const F77_REAL&, F77_REAL&);
 
   F77_RET_T
   F77_FUNC (xdbetai, XDBETAI) (const F77_DBLE&, const F77_DBLE&,
                                const F77_DBLE&, F77_DBLE&);
 
-  // ERF
-
-  F77_RET_T
-  F77_FUNC (xerf, XERF) (const F77_REAL&, F77_REAL&);
-
-  F77_RET_T
-  F77_FUNC (xderf, XDERF) (const F77_DBLE&, F77_DBLE&);
-
-  // ERFC
-
-  F77_RET_T
-  F77_FUNC (xderfc, XDERFC) (const F77_DBLE&, F77_DBLE&);
-
-  F77_RET_T
-  F77_FUNC (xerfc, XERFC) (const F77_REAL&, F77_REAL&);
-
-  // GAMMA
-
-  F77_RET_T
-  F77_FUNC (xgamma, XGAMMA) (const F77_REAL&, F77_REAL&);
-
-  F77_RET_T
-  F77_FUNC (xdgamma, XDGAMMA) (const F77_DBLE&, F77_DBLE&);
-
   // GAMMAINC
 
   F77_RET_T
   F77_FUNC (xgammainc, XGAMMAINC) (const F77_DBLE&, const F77_DBLE&,
                                    F77_DBLE&);
 
   F77_RET_T
   F77_FUNC (xsgammainc, XSGAMMAINC) (const F77_REAL&, const F77_REAL&,
                                      F77_REAL&);
 
-  // LGAMS
-
-  F77_RET_T
-  F77_FUNC (algams, ALGAMS) (const F77_REAL&, F77_REAL&, F77_REAL&);
-
-  F77_RET_T
-  F77_FUNC (dlgams, DLGAMS) (const F77_DBLE&, F77_DBLE&, F77_DBLE&);
-
   // PCHIM
 
   F77_RET_T
   F77_FUNC (dpchim, DPCHIM) (const F77_INT& n, const F77_DBLE *x,
                              const F77_DBLE *f, F77_DBLE *d,
                              const F77_INT& incfd,
                              F77_INT& ierr);
 
